### PR TITLE
Backports 0.18 pr5

### PR DESCRIPTION
**Pull request by Munkybooty**

### CODE REVIEW

File: src/txmempool.cpp [hunk 1]
```diff
@@ -1069,6 +1069,7 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
                 const CTransaction& tx2 = it2->GetTx();
                 assert(tx2.vout.size() > txin.prevout.n && !tx2.vout[txin.prevout.n].IsNull());
                 fDependsWait = true;
+                setParentCheck.insert(it2);
```
Comment (UdjinM6) [inline src/txmempool.cpp:1072]: (13767) this looks incomplete

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by UdjinM6** [comment]
13767 is not ok: should not create `descriptor_tests.cpp` and some changes in `src/txmempool.cpp` are missing

**Review by UdjinM6 — APPROVED** [review]
utACK

File: src/fs.h [hunk 2]
```diff
@@ -21,7 +21,6 @@ namespace fs = boost::filesystem;
 /** Bridge operations to C stdio */
 namespace fsbridge {
     FILE *fopen(const fs::path& p, const char *mode);
-    FILE *freopen(const fs::path& p, const char *mode, FILE *stream);
```
Comment (PastaPastaPasta) [inline src/fs.h:24]: 13159, why do we not need the changes in logging.cpp

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**